### PR TITLE
Add Multi-Suite/Label Support to TIAF Specialized Runtimes

### PR DESCRIPTION
**Pull request by jonawals**
Signed-off-by: John <jonawals@amazon.com>

## What does this PR do?

This PR implements the specializedTIAF runtime-side functionality to:
-  Support multiple suites for drawing tests from for TIAF test runs
-  Support excluding tests based on test labels for TIAF test runs

_Please describe any testing performed._

These changes have been tested with manual end-to-end testing.

### CODE REVIEW

File: Code/Tools/TestImpactFramework/Frontend/Console/Native/Code/Source/TestImpactConsoleMain.cpp [hunk 1]
```diff
@@ -48,13 +48,14 @@ namespace TestImpact::Console
...
 
             std::cout << "Constructing in-memory model of source tree and test coverage for test suite ";
-            std::cout << SuiteTypeAsString(options.GetSuiteFilter()).c_str() << ", this may take a moment...\n";
+            std::cout << SuiteSetAsString(options.GetSuiteSet()).c_str() << ", this may take a moment...\n";
             NativeRuntime runtime(
                 NativeRuntimeConfigurationFactory(ReadFileContents<CommandLineOptionsException>(options.GetConfigurationFilePath())),
                 options.GetDataFilePath(),
                 options.GetPreviousRunDataFilePath(),
                 options.GetExcludedTests(),
-                options.GetSuiteFilter(),
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Frontend/Console/Native/Code/Source/TestImpactConsoleMain.cpp:57]: This will be refactored in another PR to pass in an instance of the options.

File: Code/Tools/TestImpactFramework/Frontend/Console/Python/Code/Source/TestImpactConsoleMain.cpp [hunk 2]
```diff
@@ -39,13 +39,14 @@ namespace TestImpact::Console
...
 
             std::cout << "Constructing in-memory model of source tree and test coverage for test suite ";
-            std::cout << SuiteTypeAsString(options.GetSuiteFilter()).c_str() << ", this may take a moment...\n";
+            std::cout << SuiteSetAsString(options.GetSuiteSet()).c_str() << ", this may take a moment...\n";
             PythonRuntime runtime(
                 PythonRuntimeConfigurationFactory(ReadFileContents<CommandLineOptionsException>(options.GetConfigurationFilePath())),
                 options.GetDataFilePath(),
                 options.GetPreviousRunDataFilePath(),
                 options.GetExcludedTests(),
-                options.GetSuiteFilter(),
```
Comment (jonawals) [inline Code/Tools/TestImpactFramework/Frontend/Console/Python/Code/Source/TestImpactConsoleMain.cpp:48]: This will be refactored in another PR to pass in an instance of the options.

File: Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/Artifact/Factory/TestImpactPythonTestTargetMetaMapFactory.cpp [hunk 3]
```diff
@@ -61,12 +64,31 @@ namespace TestImpact
...
             const auto testSuites = test[Keys[TestSuitesKey]].GetArray();
+            bool skipTest = false;
             for (const auto& suite : testSuites)
             {
                 // Check to see if this test target has the suite we're looking for
                 if (const auto suiteName = suite[Keys[SuiteKey]].GetString();
-                    strcmp(SuiteTypeAsString(suiteType).c_str(), suiteName) == 0)
+                    suiteSet.contains(suiteName))
                 {
+                    const auto suiteLabels = suite[Keys[SuiteLabelsKey]].GetArray();
```
Comment (greerdv) [inline Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/Artifact/Factory/TestImpactPythonTestTargetMetaMapFactory.cpp:74]: this block of code looks identical to one above, is it possible to make this a reusable utility function?

File: Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/TestEngine/Python/TestImpactPythonTestEngine.cpp [hunk 4]
```diff
@@ -173,14 +172,24 @@ namespace TestImpact
                 testTargetTimeout,
                 globalTimeout,
                 callback,
-                std::nullopt),
-            integrityFailurePolicy);
+                std::nullopt);
+
+            if(const auto integrityErrors = GenerateIntegrityErrorString(result);
```
Comment (greerdv) [inline Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/TestEngine/Python/TestImpactPythonTestEngine.cpp:177]: This block seems to be repeated a bunch of times as well, could it be a function?

**Review by greerdv — APPROVED** [review]

File: Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/TestEngine/Python/TestImpactPythonTestEngine.cpp (hunk 4 above)
Comment (jonawals) [reply Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/TestEngine/Python/TestImpactPythonTestEngine.cpp:177]: This particular repetition is annoying and there is no simple refactor for the time being as the test runners are different types. It's on the radar and will be addressed at a later date 👍

File: Code/Tools/TestImpactFramework/Runtime/Native/Code/Include/TestImpactFramework/Native/TestImpactNativeRuntime.h [hunk 5]
```diff
@@ -30,7 +30,8 @@ namespace TestImpact
         //! @param dataFile The optional data file to be used instead of that specified in the config file.
         //! @param previousRunDataFile The optional previous run data file to be used instead of that specified in the config file.
         //! @param testsToExclude The tests to exclude from the run (will override any excluded tests in the config file).
-        //! @param suiteFilter The test suite for which the coverage data and test selection will draw from.
+        //! @param suiteSet The test suites for which the coverage data and test selection will draw from.
```
[the remark this answers is not in the document]
Comment (jonawals) [reply Code/Tools/TestImpactFramework/Runtime/Native/Code/Include/TestImpactFramework/Native/TestImpactNativeRuntime.h:33]: I've changed it to "from" 👍

File: Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/Artifact/Factory/TestImpactPythonTestTargetMetaMapFactory.cpp (hunk 3 above)
Comment (hultonha) [reply Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/Artifact/Factory/TestImpactPythonTestTargetMetaMapFactory.cpp:74]: +1

**Review by hultonha — APPROVED** [review]
LGTM just a couple of minor comments

**Review by moraaar — APPROVED** [review]

File: Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/Artifact/Factory/TestImpactPythonTestTargetMetaMapFactory.cpp (hunk 3 above)
Comment (jonawals) [reply Code/Tools/TestImpactFramework/Runtime/Python/Code/Source/Artifact/Factory/TestImpactPythonTestTargetMetaMapFactory.cpp]: Fixed 👍

File: Code/Tools/TestImpactFramework/Runtime/Native/Code/Source/TestEngine/Native/TestImpactNativeTestEngine.cpp [hunk 6]
```diff
@@ -203,7 +202,19 @@ namespace TestImpact
                 testTargetTimeout,
                 globalTimeout,
                 callback,
-                AZStd::nullopt),
-            integrityFailurePolicy);
+                AZStd::nullopt);
+
+            if(const auto integrityErrors = GenerateIntegrityErrorString(result);
```
[the remark this answers is not in the document]
Comment (jonawals) [reply Code/Tools/TestImpactFramework/Runtime/Native/Code/Source/TestEngine/Native/TestImpactNativeTestEngine.cpp:229]: Added an API comment 👍
